File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>>mp;
        for(int i=0;i<strs.size();i++){
            vector<char>freq(26, 0);
            for(char c: strs[i]){
                freq[c - 'a']++;
            }
            string key = "";
            for(int i=0;i<26;i++){
                key += to_string(freq[i]) + '#';
            }
            
            mp[key].push_back(strs[i]);
        }
        vector<vector<string>>ans;
        for(auto it: mp){
            ans.push_back(it.second);
        }
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        map = defaultdict(list)
        for i in strs:
            map[tuple(sorted(i))].append(i)
        ans = []
        for i in map.values():
            ans.append(i)
        return ans